src/java/davmail/util/SystemdNotify.java

```diff
@@ -0,0 +1,106 @@
+// Based on this answer https://stackoverflow.com/a/36945256/784804
+
+/*
+ * Copyright (c) 2018 Ian Kirk
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package davmail.util;
+
+import com.sun.jna.Library;
+import com.sun.jna.Native;
+
+public final class SystemdNotify {
+
+  interface SystemD extends Library {
+
+    final SystemD INSTANCE = Native.load(SYSTEMD_SO, SystemD.class);
+
+    int sd_notify(int unset_environment, String state);
+  }
+
+  private static final String SYSTEMD_SO = "systemd";
+  private static final String READY      = "READY=1";
+  private static final String RELOADING  = "RELOADING=1";
+  private static final String STOPPING   = "STOPPING=1";
+  private static final String STATUS     = "STATUS=%s";
+  private static final String WATCHDOG   = "WATCHDOG=1";
+  private static final String MAINPID    = "MAINPID=%d";
+  private static final String ERRNO      = "ERRNO=%d";
+  private static final String BUSERROR   = "BUSERROR=%s";
+  private static String       osType;
+
+  public static void busError(final String error) {
+    notify(String.format(BUSERROR, error));
+  }
+
+  public static void errno(final int errno) {
+    notify(String.format(ERRNO, errno));
+  }
+
+  public static void mainPid(final long pid) {
+    notify(String.format(MAINPID, pid));
+  }
+
+  public static void notify(final String message) {
+
+    if (!osType().startsWith("linux"))
```

Comment:
a) instead of a platform check I think just attempting to load the systemd lib and the notify call would be more compatible, assuming proper error handling is performed. e.g. a linkage errors and certain exit code from system should be handled without propagating an exception.

b) the process might run on linux and even on a systemd enabled machine, but might still not be systemd-managed (e.g. someone running it on-demand in a shell). how does systemd respond in in this case?

Reply:
(b) is how I use davmail. I just run davmail from the command line and don't start it using systemd. I would prefer to not lose that capability.

Reply:
When launched without systemd, davmail works just fine with this patch: `sd_notify()` return code is `0` if _`$NOTIFY_SOCKET` was not set and hence no status message could be sent_ and `notify()` simply returns.
I'll try to work out handling the absence of the library rather that platform check, maybe catching `java.lang.UnsatisfiedLinkError`.

Reply:
Sounds good, @niol